### PR TITLE
Chore: Remove bus.Dispatch from some login packages

### PR DESCRIPTION
**Pull request by zserge**
Related to #44130

### CODE REVIEW

File: pkg/api/http_server.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/infra/remotecache"
 	"github.com/grafana/grafana/pkg/infra/tracing"
+	loginpkg "github.com/grafana/grafana/pkg/login"
```
Comment (sakjur) [inline pkg/api/http_server.go:24]: Is `login` overloaded?

File: pkg/api/login_test.go [hunk 2]
```diff
@@ -740,5 +740,5 @@ func mockAuthenticateUserFunc(user *models.User, authmodule string, err error) {
 	}
 }
 func resetAuthenticateUserFunc() {
-	login.AuthenticateUserFunc = login.AuthenticateUser
+	// login.AuthenticateUserFunc = login.AuthenticateUser
```
Comment (sakjur) [inline pkg/api/login_test.go:743]: Fix this before merge: in or out, let's not leave commented out code, at least not without a comment about it 🙂

File: pkg/login/auth.go [hunk 3]
```diff
@@ -25,14 +27,34 @@ var (
...
+	AuthenticateUser(context.Context, *models.LoginUserQuery) error
+}
+
+var AuthenticateUserFunc = func(context.Context, *models.LoginUserQuery) error {
+	panic("AuthenticateUserFunc is not implemented")
+}
 
 func Init() {
-	bus.AddHandler("auth", AuthenticateUser)
+}
```
Comment (sakjur) [inline pkg/login/auth.go:39]: Remove?

File: pkg/middleware/middleware_basic_auth_test.go [hunk 4]
```diff
@@ -78,6 +79,7 @@ func TestMiddlewareBasicAuth(t *testing.T) {
 		const password = "MyPass"
 		const salt = "Salt"
 
+		login.ProvideService(sc.sqlStore, &logintest.LoginServiceFake{}) // XXX hack
```
Comment (sakjur) [inline pkg/middleware/middleware_basic_auth_test.go:82]: Keep or replace with fake?

File: pkg/server/server.go [hunk 5]
```diff
@@ -119,6 +119,7 @@ func (s *Server) init() error {
 	}
 
 	login.Init()
+	login.ProvideService(s.HTTPServer.SQLStore, s.HTTPServer.Login) // XXX hack
```
Comment (sakjur) [inline pkg/server/server.go:122]: Keep or replace with fake?

File: pkg/services/login/logintest/logintest.go [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+	ExpectedError        error
+}
+
+func (a *AuthInfoServiceFake) LookupAndUpdate(ctx context.Context, query *models.GetUserByAuthInfoQuery) (*models.User, error) {
+	a.LatestUserID = query.UserId
+	return a.ExpectedUser, a.ExpectedError
+}
+
+func (a *AuthInfoServiceFake) GetAuthInfo(ctx context.Context, query *models.GetAuthInfoQuery) error {
+	fmt.Println("==== GET AUTH INFO === ", query, a.ExpectedError)
```
Comment (sakjur) [inline pkg/services/login/logintest/logintest.go:37]:
```suggestion
```

**Review by sakjur — COMMENTED** [review]
Some development artifacts are left in, can you take a pass over this and adress them?

File: pkg/api/http_server.go (hunk 1 above)
Comment (sakjur) [reply pkg/api/http_server.go:24]: (I think the answer is yes, so no action required, just press resolve if it is)

File: pkg/login/auth.go [hunk 7]
```diff
@@ -25,14 +27,34 @@ var (
 
 var loginLogger = log.New("login")
 
-var AuthenticateUserFunc = AuthenticateUser
+type Authenticator interface {
+	AuthenticateUser(context.Context, *models.LoginUserQuery) error
+}
+
+var AuthenticateUserFunc = func(context.Context, *models.LoginUserQuery) error {
```
Comment (spinillos) [inline pkg/login/auth.go:34]: Do we need this? I'm checking the code and I found an occurrence inside `login.go` file and we can change directly with the authenticator call.

File: pkg/server/server.go (hunk 5 above)
Comment (spinillos) [reply pkg/server/server.go:122]: I think that this provider should be inside `background_services.go` file like the rest of services that are not called directly from the injector system.

File: pkg/login/auth.go (hunk 3 above)
Comment (spinillos) [reply pkg/login/auth.go:39]: Yes, the Init is empty. If we want to keep it it should be called inside the Provide function like we are doing in other services.

File: pkg/server/server.go (hunk 5 above)
Comment (sakjur) [reply pkg/server/server.go]: 🤦 I thought this was a test.

Do not replace with a fake 😅

File: pkg/services/login/loginservice/loginservice.go [hunk 8]
```diff
@@ -130,6 +130,44 @@ func (ls *Implementation) UpsertUser(ctx context.Context, cmd *models.UpsertUser
...
+				"Error disabling external user",
+				"user",
+				userQuery.Result.Login,
+				"message",
+				err.Error(),
+			)
+			return err
+		}
+	}
+	return nil
```
Comment (sakjur) [inline pkg/services/login/loginservice/loginservice.go:168]:
```suggestion
	if userInfo.IsDisabled {
		return nil
	}
	
	logger.Debug(
		"Disabling external user",
		"user",
		userQuery.Result.Login,
	)

	// Mark user as disabled in grafana db
	disableUserCmd := &models.DisableUserCommand{
		UserId:     userQuery.Result.UserId,
		IsDisabled: true,
	}

	if err := ls.SQLStore.DisableUser(ctx, disableUserCmd); err != nil {
		logger.Debug(
			"Error disabling external user",
			"user",
			userQuery.Result.Login,
			"message",
			err.Error(),
		)
		return err
	}
	return nil
```

**Review by sakjur — APPROVED** [review]

**Pull request merged**